api/server.go

```diff
@@ -62,7 +68,7 @@ type Server struct {
 	listener   net.Listener
 }
 
-func NewServer(ctx context.Context) (*Server, error) {
+func NewServer(ctx context.Context, kataEnabledFlag bool) (*Server, error) {
```

Comment:
Would be nice to create a `ServerOptions` struct that is instead passed to the `NewServer` function. This would allow for more flexibility in the future. For example, if we wanted to add a `MaxConnections` field to the `ServerOptions` struct, we could do so without breaking the signature.

Also, it would be better to name the flag `vsockEnabled` instead as there's nothing Kata specific about it. In future, we might want to add support for other VM/runtimes that use vsock.